### PR TITLE
Adding DNS Resolver module

### PR DESCRIPTION
**Pull request by danhuss**
I pulled the majority of this from a fork by @chris-cyliu . The biggest changes from there were:

- Documentation... 
- Renamed module to pfsense_dns_resolver in case a separate DNS Forwarder module comes in later.
- Added a bunch more settings that I had in my config file.

I ran into some issues setting up a test VM (not the greatest with networking), so it could probably use a little more. I also missed some settings when I compared them to the GUI, but have them noted in the code. Any feedback or suggestions are very welcome.

related to: #17

### CODE REVIEW

File: plugins/modules/pfsense_dns_resolver.py [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: pfsense_dns_resolver
+version_added: 0.5.0
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:18]: Bump to 0.6.0 now.

File: plugins/modules/pfsense_dns_resolver.py [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+
+DNS_HOST_SPEC = dict(
+    host=dict(required=True, type='str'),
+    domain=dict(required=True, type='str'),
+    ip=dict(required=True, type='str'),
+    descr=dict(default="", type='str'),
+    aliases=dict(default=[], type='list', elements='dict', options=DNS_HOST_ALIAS_SPEC),
+)
+
+dns_ARGUMENT_SPEC = dict(
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:325]: Please all caps

File: plugins/modules/pfsense_dns_resolver.py [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+    infra_cache_numhosts=dict(default=10000, type='int', choices=[1000, 5000, 10000, 20000, 50000, 100000, 200000]),
+    unwanted_reply_threshold=dict(default="disabled", type='str', choices=["disabled", "5000000", "10000000", "20000000", "40000000", "50000000"]),
+    log_verbosity=dict(default=1, type='int', choices=[0, 1, 2, 3, 4, 5])
+    # TODO: Disable Auto-added Access Control
+    # TODO: Disable Auto-added Host Entries
+    # TODO: Experimental Bit 0x20 Support
+    # TODO: DNS64 Support
+)
+
+dns_REQUIRED_IF = []
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:379]: Please all caps

File: plugins/modules/pfsense_dns_resolver.py [hunk 4]
```diff
@@ -0,0 +1,573 @@
...
+
+
+class PFSenseDNSModule(PFSenseModuleBase):
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:382]: I'm wondering if this shouldn't be DNSResolverModule and DNS_RESOLVER_ for the vars.

**Review by opoplawski — CHANGES_REQUESTED** [review]

**Comment by opoplawski** [comment]
Thank you for the submission!

File: plugins/modules/pfsense_dns_resolver.py (hunk 1 above)
Comment (danhuss) [reply plugins/modules/pfsense_dns_resolver.py]: Done

File: plugins/modules/pfsense_dns_resolver.py (hunk 2 above)
Comment (danhuss) [reply plugins/modules/pfsense_dns_resolver.py]: 👍

File: plugins/modules/pfsense_dns_resolver.py (hunk 3 above)
Comment (danhuss) [reply plugins/modules/pfsense_dns_resolver.py]: good to go.

File: plugins/modules/pfsense_dns_resolver.py (hunk 4 above)
Comment (danhuss) [reply plugins/modules/pfsense_dns_resolver.py]: This makes sense. I changed the naming in this file.

File: plugins/modules/pfsense_dns_resolver.py [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+    description: Listen Port
+    required: false
+    default: null
+    type: int
+  enablessl:
+    description: Enable SSL/TLS Service
+    required: false
+    default: false
+    type: bool
+  sslcertdescr:
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:40]: We tend to use just `cert` or `certificate` elsewhere, so let's rename this to `sslcert`.

File: plugins/modules/pfsense_dns_resolver.py [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+    description: Description of the server certificate to use for SSL/TLS service.
+    required: false
+    default: ""
+    type: str
+  tlsport:
+    description: SSL/TLS Listen Port
+    required: false
+    default: null
+    type: int
+  list_active_interface_descr:
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:50]: This should just be `active_interface`.

File: plugins/modules/pfsense_dns_resolver.py [hunk 7]
```diff
@@ -0,0 +1,573 @@
...
+    description: Interface IPs used by the DNS Resolver for responding to queries from clients.
+    required: false
+    default: [ "all" ]
+    type: list
+    elements: str
+  list_outgoing_interface_descr:
```
Comment (opoplawski) [inline plugins/modules/pfsense_dns_resolver.py:56]: This should just be `outgoing_interface`.

**Review by opoplawski — CHANGES_REQUESTED** [review]
I think just these last changes and we should be good.  Thanks again for working on this.

File: plugins/modules/pfsense_dns_resolver.py (hunk 5 above)
Comment (danhuss) [reply plugins/modules/pfsense_dns_resolver.py]: It's been a little while, but I think all three of these were already different from the XML tags they were written to. So a simple find and replace all should do the trick.

**Review by opoplawski — APPROVED** [review]

**Comment by opoplawski** [comment]
Thank you for the contribution!

**Comment by opoplawski** [comment]
BTW - in the future, please use a separate branch instead of master to submit PRs from.

**Pull request merged**